src/logic/calculate.js

```diff
@@ -0,0 +1,34 @@
+import operate from './operate';
+
+const calculate = (calculatorData, buttonName) => {
+  const dataUpdate = { ...calculatorData };
+
+  if (buttonName === 'AC') {
+    dataUpdate.total = null;
+    dataUpdate.next = null;
+    dataUpdate.operation = null;
+  }
+
+  if (buttonName === '+/-') {
```

Comment:
- [ ] What if the buttonName is a number? Also what if the buttonName is a number and there is an `operation`, how would you handle this?
- [ ] What if the buttonName is a `.`?